ksqldb-functional-tests/src/test/resources/query-validation-tests/upgrades.json

```diff
@@ -201,9 +214,79 @@
         "CREATE TABLE bar AS SELECT * FROM foo;",
         "CREATE OR REPLACE TABLE bar AS SELECT * FROM foo WHERE col1 = '123';"
       ],
+      "inputs": [],
+      "outputs": []
+    },
+    {
+      "name": "DML - TABLE - add column",
+      "properties": {
+        "ksql.create.or.replace.enabled": true
+      },
+      "statements": [
+        "CREATE TABLE foo (id VARCHAR PRIMARY KEY, col1 VARCHAR, col2 VARCHAR) WITH (kafka_topic='foo', value_format='JSON');",
+        "CREATE TABLE bar AS SELECT id, col1 FROM foo;",
+        "CREATE OR REPLACE TABLE bar AS SELECT id, col1, col2 FROM foo;"
+      ],
+      "inputs": [],
+      "outputs": []
+    },
```

Comment:
Do you have plans to extend the testing done around upgrades?

It seems to me that we need testing not only to ensure commands run, but that state isn't corrupted and the outputs are correct.

E.g. we should be pumping in some initial state, (confirming the output / schemas registered in SR / metastore state), performing the upgrade, then pumping in more state and again confirming the output, SR and metastore state. 

We likely also need tests around pull queries while upgrades are running.

Thoughts?

Reply:
I've added `ReplaceIntTest` in the interim while I figure out the best way to do this type of testing. QTT would require a major overhaul to test these upgrades, which is something that I plan to give a shot. But yes, I do plan to add a lot more testing before enabling the feature by default.

> We likely also need tests around pull queries while upgrades are running.

At the moment, pull queries will likely fail as we terminate the old query and clean up the state. I will add a better exception (though anything that materializes it's own table actually isn't even upgradeable at the moment so it's not that big of a problem in the current state).